catroid/src/main/java/org/catrobat/catroid/utils/FlashUtil.java

```diff
@@ -112,6 +124,11 @@ public static void initializeFlash() {
 
 	public static void flashOn() {
 		Log.d(TAG, "flashOn()");
+
+		if (!isAvailable()) {
+			return;
+		}
+
 		if (!CameraManager.getInstance().hasCurrentCameraFlash()) {
```

Comment:
Why was if (!CameraManager.getInstance().hasCurrentCameraFlash()) not enough?

Reply:
At a second glance - this is far from ignoring the flash brick!
If the currently active camera does not have a flash, it results in us destroying the stage - IMO the ...destroyStage() part needs to go.

Reply:
As in the Ticket Description says
> Expected Behavior: The Flash brick should simply be ignored on devices without a flash.
i think we shouldn't destroy the stage. There should just nothing happen if you try to activate flash.